public/modules/assign-roles/js/module.js

```diff
@@ -2127,10 +2127,9 @@ __webpack_require__.r(__webpack_exports__);
     };
   },
   watch: {
-    positionId: function positionId(val) {
-      if (this.role_name === null) {
-        this.role_name = this.getPositionName(val);
-      }
+    positionId: function positionId(newVal, oldVal) {
+      // If role name is empty or hasn't been changed then assign Position Name
+      return this.role_name === null || this.role_name === this.getPositionName(oldVal) ? this.role_name = this.getPositionName(newVal) : '';
```

Comment:
Can we also check if the role name is ''? If you click in the role name box before changing the position, it doesn't change as `this.role_name` = '' not null